android/app/src/main/java/app/organicmaps/routing/NavigationService.java

```diff
@@ -137,6 +138,11 @@ public static NotificationCompat.Builder getNotificationBuilder(@NonNull Context
     final PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, contentIntent,
         PendingIntent.FLAG_CANCEL_CURRENT | FLAG_IMMUTABLE);
 
+    final Intent exitIntent = new Intent(context, NavigationService.class);
+    exitIntent.setAction(STOP_NAVIGATION);
+    final PendingIntent exitPendingIntent = PendingIntent.getService(context, 0, exitIntent,
+        PendingIntent.FLAG_UPDATE_CURRENT | FLAG_IMMUTABLE);
```

Comment:
What is the reasoning for selecting these two flags for intent?

Reply:
FLAG_UPDATE_CURRENT insures that there should exist only one pending intent doing same task at any moment. If there is already an existing pendingIntent which is doing same task then rather than creating new pendingIntent it updates the previous one with new one.
 
Whereas FLAG_IMMUTABLE is used for, if by any reason pendingIntent is cant be overwritten then it creates new one.